### ISSUE TITLE
create parser to detect language from git diff

### ISSUE THREAD

**Issue by CBaptiste** [algolia/algoliasearch-client-readme-generator#63]
after talk with @PLNech; this might be possible

**Comment by vvo** [comment]
What is the purpose of detecting the language?

**Comment by CBaptiste** [comment]
Right now, if person 1 and person 2 commits together, the bot will create one single commit from travis, so I need to get the commit added for a specific language.
when someone modifies the folder include/Javascript, i know the that commit is for the JS repo. However, if it is the folder include/template, I cannot know which commit modifies which language ( it is one single file)
BUT: if I get the git diff on this file, detect the "objc" from <% elsif objc? -%> that has been added, I could generate the right commit message for each language.

I don't know if it is clear :)

**Comment by PLNech** [comment]
It would avoid creating a pull request on language A with commit messages of language B, as you can see here:

![PR #85 on algolia/algoliasearch-client-android](https://cloud.githubusercontent.com/assets/1821404/15541997/2e85d230-228f-11e6-8d11-e75bf8ef93b8.png)

**Comment by CBaptiste** [comment]
Yes it happened because someone from the last time someone modified the android READMe, the the include/template has been changed as well even though it was for another language.
Hence, this ticket :)

**Comment by vvo** [comment]
Depending on the difficulty and stability of this proposal, it might not be necessary to add it (= if it takes a lot of effort to add this and is subject to bugs/inconsistencies then it may not be worth it)

**Comment by CBaptiste** [comment]
I want to wait for the tool to be stable the way it is now (getting closer)
and then I will try, and see if it is easy or not

**Comment by ercolanelli-leo** [comment]
If it's just about checking the existence of a line in the git diff, isn't 
```
git diff | grep '<%elsif objc? %>'  > /dev/null
if $?; then
   # Commit...
fi 
```
somewhat sufficient?

**Comment by CBaptiste** [comment]
I like that ! I ll give it a try !
